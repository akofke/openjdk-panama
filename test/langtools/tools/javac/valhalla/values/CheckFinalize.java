/*
 * @test /nodynamiccopyright/
 * @summary Value types do not support finalize
 *
 * @compile/fail/ref=CheckFinalize.out -XDenableValueTypes -XDrawDiagnostics CheckFinalize.java
 */

final __ByValue class CheckFinalize {
    @Override
    protected void finalize() {} // <-- error

    final __ByValue class CheckFinalizeInner {}

    void foo(CheckFinalizeInner cfi, CheckFinalize cf) {
        cfi.finalize();          // Error
        cf.finalize();           // OK.
    }
}
