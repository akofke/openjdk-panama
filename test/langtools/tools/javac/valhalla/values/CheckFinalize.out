CheckFinalize.java:9:5: compiler.err.method.does.not.override.superclass
CheckFinalize.java:15:12: compiler.err.value.does.not.support: finalize
2 errors
